src/SigSpec.Json.Middleware/SigSpecUIBuilderExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using SigSpec.Core;
+using System;
+using System.Collections.Generic;
+
+namespace SigSpec.Middleware
+{
+    public static class SigSpecBuilderExtensions
+    {
+        public static IApplicationBuilder UseSigSpec(this IApplicationBuilder app, IEnumerable<Type> hubs, Action<SigSpecGeneratorSettings> setupAction = null)
```

Comment:
Also use only two params (this and action) and add the hubs config as property to a subclass of SigSpecGeneratorSettings, see

https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.AspNetCore/Extensions/NSwagApplicationBuilderExtensions.cs#L23